connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/ChannelBuilder.scala

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connect.client
+
+import scala.language.existentials
+
+import io.grpc.{ChannelCredentials, Grpc, ManagedChannel, ManagedChannelBuilder}
+
+import org.apache.spark.sql.connect.client.SparkConnectClient.MetadataHeaderClientInterceptor
+import org.apache.spark.sql.connect.common.config.ConnectCommon
+
+class ChannelBuilder() {
```

Comment:
Add scaladoc.

(I understand it's not present consistently in the existing code base but we should have it for all public APIs. Let's add for the ones being introduced here)